main/src/org/jetbrains/bsp/bazel/install/Install.java

```diff
@@ -0,0 +1,214 @@
+package org.jetbrains.bsp.bazel.install;
+
+import ch.epfl.scala.bsp4j.BspConnectionDetails;
+import com.google.common.base.Splitter;
+import com.google.gson.GsonBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.jetbrains.bsp.bazel.common.Constants;
+
+public class Install {
+  public static final String INSTALLER_BINARY_NAME = "bazelbsp-install";
+  public static final String SERVER_CLASS_NAME = "org.jetbrains.bsp.bazel.server.Server";
+
+  private static final String DEBUGGER_SHORT_OPT = "x";
+  private static final String JAVA_SHORT_OPT = "j";
+  private static final String BAZEL_SHORT_OPT = "b";
+  private static final String HELP_SHORT_OPT = "h";
+  private static final String DIRECTORY_SHORT_OPT = "d";
+
+  public static void main(String[] args) throws IOException {
+    // This is the command line which will be used to start the BSP server. See:
+    // https://build-server-protocol.github.io/docs/server-discovery.html#build-tool-commands-to-start-bsp-servers
+    List<String> argv = new ArrayList<>();
+
+    Options cliOptions = getCliOptions();
+    CommandLineParser parser = new DefaultParser();
+    HelpFormatter formatter = new HelpFormatter();
+    formatter.setWidth(120);
+    boolean hasError = false;
+    PrintWriter writer = new PrintWriter(System.err);
+    try {
+      CommandLine cmd = parser.parse(cliOptions, args, false);
+      if (cmd.hasOption(HELP_SHORT_OPT)) {
+        formatter.printHelp(INSTALLER_BINARY_NAME, cliOptions);
+      } else {
+        // Add Java binary
```

Comment:
i think that extracting this section to the method `addJavaBinary` would help to remove this comment.